orttraining/orttraining/python/training/ortmodule/_training_manager.py

```diff
@@ -100,6 +103,26 @@ def forward(self, *inputs, **kwargs):
                 if self._loglevel <= _logger.LogLevel.WARNING:
                     warnings.warn("Fast path enabled - skipping checks for rebuilding gradient graph, execution agent creation, and device during training.",
                                   UserWarning)
+        # create cache object
+        if self._enable_grad_acc_optimization:
+            if self._cache == None:
+                self._cache = C.OrtValueCache()
+
+            # parse param versions to detect change or no change
+            if self._param_version_map == None:
+                self._param_version_map = dict()
+
+            for name, param in self._flattened_module.named_parameters():
+                if name in self._graph_info.frontier_node_arg_map:
+                    arg_name = self._graph_info.frontier_node_arg_map[name]
+                    if name not in self._param_version_map:
+                        self._param_version_map[name] = param._version
+                    elif param._version != self._param_version_map[name]:
+                        # there is an updated param, so remove entry from cache
+                        # in order to recompute the value
+                        if self._cache.count(arg_name):
+                            self._cache.remove(arg_name)
+                        self._param_version_map[name] = param._version
```

Comment:
How about creating a `GradientAccumulationManager` class, initialize it on `GraphExecutionManager` and this class handles the whole thing, making it more decoupled?